### ISSUE TITLE
UML needs corrections

### ISSUE THREAD

**Issue by elf-pavlik** [valueflows/agent#65]
![uml](https://raw.githubusercontent.com/valueflows/agent/e67cf6b099b91a58f313507b27893b4cf82ac18f/assets/agent.png)

Current diagram seems to show _vf:memberOf_ and _vf:hasMember_ as subclasses of _rdf:Property_. We define them as **instances** of _rdf:Property_. You could draw them as arrows between _vf:Group_ and _vf:Person_.

Also _vf:Relationship_ as a class doesn't use _owl:inveseOf_ it only makes sense on _rdf:Property_ in this diagram.

**Comment by fosterlynn** [comment]
> You could draw them as arrows between vf:Group and vf:Person.

vf:Groups can be members of vf:Groups.  I'm thinking maybe we should take the member stuff off the picture, what do you all think?

Also wondering if I should make vf:Agent into foaf:Agent?  Etc.

**Comment by fosterlynn** [comment]
Actually, I'm wondering if we want to keep the UML diagrams more at a logical level, so we can eventually support non-LOD interoperability schemes?  I don't right now know what those implementations might look like.  Hmmm.

**Comment by elf-pavlik** [comment]
> vf:Groups can be members of vf:Groups. I'm thinking maybe we should take the member stuff off the picture, what do you all think?

In that case you can draw it as arrow between Agent and Group

> Also wondering if I should make vf:Agent into foaf:Agent? Etc.

:+1: also http://xmlns.com/foaf/spec/#term_member

I can make PR to JSON-LD context that will make aliases:

``` jsonld
{
  "@context": {
    "foaf": "http://xmlns.com/foaf/0.1/",
    "Agent": "foaf:Agent",
    "Person": "foaf:Person",
    "Group": "foaf:Group",
    "hasMember": { "@id": "foaf:member", "@type": "@id" },
    "memberOf": { "@reverse": "foaf:member" }
  }
}
```

This way for those who want to rely on that JSON-LD context, difference between `vf:` and `foaf:` namespaces stays hidden away.

**Comment by fosterlynn** [comment]
> I can make PR to JSON-LD context

Sounds good.  Except I'm wondering about foaf:member, I will need to look at it to see how they define it.  We also talked at one point about other types we want to define, so people can create subproperties of them and know something about their behavior.  (Will be later today I think.)

So what do you think is most useful on the UML diagram then?  foaf:Agent? Agent?

**Comment by fosterlynn** [comment]
I looked up foaf:member.  Sounds fine to me.

**Comment by elf-pavlik** [comment]
Let's always use prefixes, at least until we get close to 1.0 release :wink:

**Comment by elf-pavlik** [comment]
fixed in #67
